### PR TITLE
Fix esi parsing with turpentine

### PR DESCRIPTION
**Pull request by luigifab**
### Description

With Nexcess Magento Turpentine for Varnish, sometimes there is a strange bug with ESI processing (ESI is not processed).

Example: `window.location='https://.../fr/customer/address/delete/form_key/<esi:include src='http://.../fr/turpentine/esi/getFormKey/ttl/86400/method/esi/scope/global/access/private/' />/id/'+addressId;`

In #2682, the solution is to invert ' and ".

But in fact, this is right and wrong (this must be done from Turpentine side).
The real solution is to remove `CDATA` tags (https://stackoverflow.com/a/4504180/2980105).
I think I have tested without and added them for the PR.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by fballiano — APPROVED** [review]

**Review by kiatng — DISMISSED** [review]

**Comment by colinmollenhour** [comment]
If the CDATA markers are removed doesn't everything need to be htmlescaped? E.g. `"` should either be inside a CDATA or escaped as `&quot;`?

**Comment by luigifab** [comment]
My html validator is okay without the CDATA:

image

**Comment by elidrissidev** [comment]
AFAIK, CDATA is required only for XML parsers since they consider any appearance of `</` as a closing tag, even if it's inside quotes in JS.

Also see: https://stackoverflow.com/a/66865

**Comment by colinmollenhour** [comment]
> Also see: https://stackoverflow.com/a/66865
There are a lot of answers there and I'm still confused as to what's "best" and it probably depends on your opening tag as well..

For the purpose of keeping your PR to one issue and not conflating another I think it would be best to separate the CDATA discussion from this pull request by reverting the CDATA change so it doesn't hold up your PR. Then if CDATA should be removed we could remove them everywhere, or just leave them and not fret over it.

File: app/design/frontend/base/default/template/customer/address/book.phtml [hunk 1]
```diff
@@ -89,12 +89,10 @@
     <p class="back-link"><a href="<?php echo $this->escapeUrl($this->getBackUrl()) ?>"><small>&laquo; </small><?php echo $this->__('Back') ?></a></p>
 </div>
 <script type="text/javascript">
-//<![CDATA[
     function deleteAddress(addressId) {
```
Comment (fballiano) [inline app/design/frontend/base/default/template/customer/address/book.phtml:92]:
```suggestion
//<![CDATA[
    function deleteAddress(addressId) {
```

**Review by fballiano — APPROVED** [review]
I've re-added CDATA as per the discussion so we can approve it and merge it

**Review by ADDISON74 — APPROVED** [review]

**Comment by luigifab** [comment]
Oh my bad. The CDATA was not removed in _app/design/frontend/base/default/template/customer/address/book.phtml_!

**Pull request merged**